### ISSUE TITLE
u32 indices

### ISSUE THREAD

**Issue by erethedaybreak** [simnalamburt/obj-rs#33]
Please increase the indices to store u32, or make it generic across types. u16 is quite easy to run out of.

```
pub struct Obj<V = Vertex> {
    /// Object's name.
    pub name: Option<String>,
    /// Vertex buffer.
    pub vertices: Vec<V>,
    /// Index buffer.
    pub indices: Vec<u32>,
}
```

Perhaps mention in the readme that the corresponding OpenGL calls have to take GL_UNSIGNED_INT.

```
gl::BindBuffer(gl::ELEMENT_ARRAY_BUFFER, self.ebo);
gl::DrawElements(gl::TRIANGLES, indices.len() as i32, gl::UNSIGNED_INT, ptr::null());
```

**Comment by simnalamburt** [comment]
@erethedaybreak Good point! 

```
pub struct Obj<V = Vertex, I = u16> {
    /// Object's name.
    pub name: Option<String>,
    /// Vertex buffer.
    pub vertices: Vec<V>,
    /// Index buffer.
    pub indices: Vec<I>,
}
```

This kind of form is likely to be able to meet your needs without compromising backwards compatibility. For the time being I do not have time to fix this myself. If you fix it this way, I will merge PR. Do you have any thoughts?

**Comment by erethedaybreak** [comment]
Thanks. From what I saw, it will also affect `load_obj`, `raw::parse_obj` and the `FromRawVertex` trait.

I would rather wait until you have time to look through it, for now I simply replaced all `u16` with `u32` in a local copy.

I'm happy to test though. Very useful library, many thanks for the implementation!

I'm using this in a raw OpenGL context, i.e. not through glium.

**Comment by simnalamburt** [comment]
Closed by #36. Now we are able to use non-u16 indices!

**Comment by gambhiro** [comment]
Super!
